### ISSUE TITLE
when.timespans.start|end.in|earliest|latest

### ISSUE THREAD

**Issue by docuracy** [LinkedPasts/linked-places-format#42]
[Specification](https://github.com/LinkedPasts/linked-places-format#when-required) requires one of in|earliest|latest, but the given examples omit these properties.

**Comment by kgeographer** [comment]
?? The [large example](https://github.com/LinkedPasts/linked-places-format#an-example-linked-places-record) does use `{"in":___}` within `start` and `end` expressions for all timespans. And the example within the "when" section of the spec uses all three. Maybe I misunderstand.
```"when": {
  "timespans": [
    { "start": { "in": "yyyy-mm" },
      "end": {
          "earliest": "-yyyy",
          "latest": "yyyy-mm-dd" }
    }
  ],
  "periods": [
    { "name": "Anglo-Saxon Period, 449-1066",
      "uri": "http://n2t.net/ark:/99152/p06c6g3whtg" }
  ],
  "label": "for a century during the Anglo-Saxon period",
  "duration": "P100Y"
}
```

**Comment by docuracy** [comment]
Ah, so they do - but these smaller examples do not:

https://github.com/LinkedPasts/linked-places-format#geometry-required
https://github.com/LinkedPasts/linked-places-format#relations-optional

**Comment by kgeographer** [comment]
Yes indeed! I will repair that oversight right away. However, I have wondered whether alternate forms for the `when` object might be considered, for example flattened. Maybe deserves a separate issue.

**Comment by docuracy** [comment]
I wondered about that too, mainly to improve the human-readability/comprehension. How about allowing (in addition to the existing specification so as to preserve backward-compatibility) a single-string expression for `when` in which symbols have the following meanings:

> **>** after
> **>=** earliest
> **<** before
> **<=** latest
> **/** range delimiter _(not '-', which is used as a year-month-day delimiter in ISO 8601 expressions)_

Valid values would include:

> Between (and during) 1939 and 1945: **{"when": "1939/1945"}**
> After 2010: **{"when":">2010"}** _or_ **{"when":">=2011"}**
> Not before 1 April 1999: **{"when":">=2010-04-01"}**

**Comment by docuracy** [comment]
Hmm, I see that there could be a problem with saying something like **{"when":">1066"}**, because it's not obvious whether that refers to the start or the end of a timespan: I suggest that end dates must be preceded by the '/' delimiter, so this example would read 'started after 1066' and not 'ended after 1066'.

It might also be useful to have a simple shorthand such as '?' to indicate (optionally) when a date is unknown. 

I'll update my previous comment to account for these.

**Comment by docuracy** [comment]
Yes, I like your wariness, Ryan. However, under this proposal 'complicated needs' would still be met by the existing `when` specification: the idea is to offer an alternative to that 3-tier JSON object which can easily be grasped by anyone typing data into a spreadsheet. I know that I'm far from alone in finding the existing `when` specification the most difficult aspect of LPF when trying to wrestle data into conformity, to the point where I'm often inclined not to bother.

I suggested dropping time elements from the proposed '**un**complicated needs' format partly because I've _never_ encountered them: I'd like to know if others in this group have seen them more than very occasionally. Typing `xsd:dateTime` values into a spreadsheet is hardly uncomplicated and they're not easy on the eye, but I suppose if somebody wants to do it there's no harm in allowing it.

I realise I may be missing a bigger picture, but I don't think we'd be sacrificing any ability to sort and query on dates because in order to get .lp.json into a database it has to be parsed, and any parser might easily be configured to remove and convert `<>=` characters into their `latest|earliest|in` equivalents, and to handle the remaining ISO 8601-2 expressions too. Do we maintain a list somewhere of implementations (other than _Peripleo_) that support LPF?

**Comment by rybesh** [comment]
> I don't think we'd be sacrificing any ability to sort and query on dates because in order to get .lp.json into a database it has to be parsed, and any parser might easily be configured to remove and convert <>= characters into their latest|earliest|in equivalents, and to handle the remaining ISO 8601-2 expressions too. 

LPF is also RDF, and ideally it could be loaded into a triplestore and queried using SPARQL without having to install specialized SPARQL extensions for parsing non-standard date expressions. Even handling ISO 8601-2 is a tall order, especially at Level 2.

**Comment by docuracy** [comment]
Thanks, Vincent. If I've understood the CIDOC CRM correctly, this is more or less what Karl has implemented in the existing LPF: what I'm proposing is an optional alternative format for dates that encapsulates the wordy English expressions with internationally-understood symbols, so for example `P82a_begin_of_the_begin` becomes `>=`. My suggestion to disallow time elements applies only to this optional alternative string format for the sake of simplicity: any representation that needs greater granularity of time than just a date could still use the original standard.
